qwen_agent/memory/memory.py

```
@@ -0,0 +1,137 @@
+            if messages and messages[-1].role == USER:
+                query = extract_text_from_message(messages[-1], add_upload_info=False)
+
+            # Keyword generation
+            if query and self.rag_keygen_strategy.lower() != 'none':
+                module_name = 'qwen_agent.agents.keygen_strategies'
+                module = import_module(module_name)
+                cls = getattr(module, self.rag_keygen_strategy)
+                keygen = cls(llm=self.llm)
+                response = keygen.run([Message(USER, query)], files=rag_files)
+                last = None
+                for last in response:
```

Ruff (0.8.2)

Loop control variable `last` not used within loop body

(B007)